File: app/src/main/java/ostmodern/skylark/di/FragmentBuildersModule.java

package ostmodern.skylark.di;


import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import ostmodern.skylark.ui.detail.SetDetailFragment;
import ostmodern.skylark.ui.sets.SetListFragment;

@Module
public abstract class FragmentBuildersModule {

    @ContributesAndroidInjector(modules = {MVPPresenterModule.class, MVPViewsModule.class})
    abstract SetListFragment contributeSetListFragment();

    @ContributesAndroidInjector(modules = {MVPPresenterModule.class, MVPViewsModule.class})
    abstract SetDetailFragment contributeSetDetailFragment();

}

File: app/src/main/java/ostmodern/skylark/di/FragmentBuildersModule.kt
package ostmodern.skylark.di


import dagger.Module
import dagger.android.ContributesAndroidInjector
import ostmodern.skylark.ui.detail.SetDetailFragment
import ostmodern.skylark.ui.sets.SetListFragment

@Module
abstract class FragmentBuildersModule {

    @ContributesAndroidInjector(modules = arrayOf(MVPPresenterModule::class, MVPViewsModule::class))
    internal abstract fun contributeSetListFragment(): SetListFragment

    @ContributesAndroidInjector(modules = arrayOf(MVPPresenterModule::class, MVPViewsModule::class))
    internal abstract fun contributeSetDetailFragment(): SetDetailFragment

}
